packages/connectivity/connectivity/android/src/main/java/io/flutter/plugins/connectivity/ConnectivityBroadcastReceiver.java

```diff
@@ -30,6 +30,7 @@
   private Connectivity connectivity;
   private EventChannel.EventSink events;
   private Handler mainHandler = new Handler(Looper.getMainLooper());
+  private ConnectivityManager.NetworkCallback networkCallback;
```

Comment:
Why are we caching this?

And it looks like we're only setting it in `onCancel`, is that really right?

Reply:
Ah sorry it's inversed .